app/src/main/kotlin/com/heartlessveteran/myriad/navigation/Destinations.kt

```
@@ -0,0 +1,244 @@
+    }
+    
+    fun validateSearchQuery(query: String?): Boolean {
+        return query?.length?.let { it <= 500 } ?: true // Allow empty queries
+    }
+    
+    fun validateContentType(type: String?): Boolean {
+        return type?.let { 
+            try {
+                ContentType.valueOf(it.uppercase())
+                true
+            } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)